src/main/scala/beam/router/graphhopper/BikeGraphHopperWrapper.scala

```diff
@@ -0,0 +1,57 @@
+package beam.router.graphhopper
+
+import beam.agentsim.agents.vehicles.BeamVehicleType
+import beam.router.Modes
+import beam.router.Modes.BeamMode
+import beam.router.graphhopper.BikeGraphHopperWrapper.FastestProfileName
+import beam.router.model.BeamLeg
+import beam.sim.common.GeoUtils
+import com.graphhopper.config.Profile
+import com.graphhopper.util.Parameters
+import com.graphhopper.{GHRequest, GraphHopper, ResponsePath}
+import org.matsim.api.core.v01.{Coord, Id}
+
+import scala.collection.JavaConverters._
+
+class BikeGraphHopperWrapper(
+  graphDir: String,
+  geo: GeoUtils,
+  id2Link: Map[Int, (Coord, Coord)]
+) extends GraphHopperWrapper(graphDir, geo, id2Link) {
+
+  override protected val beamMode: Modes.BeamMode = BeamMode.BIKE
+
+  override def getProfile(): Profile = BikeGraphHopperWrapper.getFastestProfile
+
+  override def createGraphHopper(): GraphHopper = {
+    new GraphHopper()
+  }
+
+  override def prepareRequest(request: GHRequest): Unit = {
+    request.setProfile(FastestProfileName)
+    request.setPathDetails(Seq(Parameters.Details.EDGE_ID, Parameters.Details.TIME).asJava)
+  }
+
+  override protected def getLinkTravelTimes(responsePath: ResponsePath, totalTravelTime: Int): IndexedSeq[Double] = {
+    responsePath.getPathDetails
+      .asScala(Parameters.Details.TIME)
```

Comment:
I don't have a compiler while checking, but why two `asScala`'s?

Reply:
The first one for converting the Map, the second one for converting the values of that map.